### PR TITLE
chore: Remove promise manager usage from ingestion warnings & db

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Promise manager adds confusion to how things work and could be causing the ingestion pipeline to stall at points, lets nuke it.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by bretthoerner** [comment]
Interesting, I hadn't run into `PromiseManager` yet.

Behaviorally the diff makes sense to me, but it does seem like we'll now block and `await` a Kafka produce for every `captureIngestionWarning` call, rather than just fire-and-forget it in the background. Won't this only add to ingestion stalls?

**Comment by tiina303** [comment]
> Behaviorally the diff makes sense to me, but it does seem like we'll now block and await a Kafka produce for every captureIngestionWarning call, rather than just fire-and-forget it in the background. Won't this only add to ingestion stalls?

yes, potentially, but our ingestion parallelism should handle this, no? Also afaik there's two options: someone somewhere waits for it or we drop it occasionally. Wouldn't be surprised if we did the latter.

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ export async function eachBatchParallelIngestion(
                     const team = await queue.pluginsServer.teamManager.getTeamForEvent(currentBatch[0].pluginEvent)
                     const distinct_id = currentBatch[0].pluginEvent.distinct_id
                     if (team && WarningLimiter.consume(`${team.id}:${distinct_id}`, 1)) {
-                        captureIngestionWarning(queue.pluginsServer.db, team.id, 'ingestion_capacity_overflow', {
+                        await captureIngestionWarning(queue.pluginsServer.db, team.id, 'ingestion_capacity_overflow', {
```
Comment (xvello) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts:93]: ditto:
> nit: alternatively, we can push the promise into processingPromises to be awaited at the end of the batch

**Review by xvello — APPROVED** [review]
Yeah, let's nuke this manager 🔥 

`nit for possibly later`: I'm wondering whether we really need the delivery guarantee though: `await`ing ensures we never lose an ingestion warning on pod crash, but do we really need this guarantee? I feel like we could get away with firing off `captureIngestionWarning` and not `await`ing it. 

`captureIngestionWarning` returns once the message is in the rdkafka producer queue, that we expect to be properly flushed on graceful shutdown (we await `kafkaProducer.disconnect()` on `closeHub`), so the only data loss scenario is pod crash/oom.

**Comment by tiina303** [comment]
There was something about voided promises that was the reason, promiseManager was added in the first place iirc ... too many voided promises cause some crashes / stalling 🤷‍♀️

**Pull request merged**